unit_tests/test_aggregator.py

```diff
@@ -298,12 +280,14 @@ def test_execute_batch_commands_execute_batch_called(aggregator):
             mocked_method.assert_called()
 
     @staticmethod
+    @pytest.mark.usefixtures("mock_execute_batch_command_methods")
```

Comment:
Why are these needed now? Before the refactoring into a single class they weren't here. Can you please explain?

Reply:
These were already there but placed differently. They were placed at the beginning of each corresponding class. The fixtures were then applied to all the test methods in that class. As it was separated in multiple classes, it was possible to apply these in such way to avoid this slight repetition you point out. 
Now that all the test methods are in a single class, the approach is to pass the needed fixtures explicitly to each one of them.